src/main/proto/wfa/measurement/api/v2alpha/exchange_workflow.proto

```diff
@@ -37,15 +37,44 @@ message ExchangeWorkflow {
     // the step-specific name to a label for the input. The step is ready to run
     // when, for each input label, there is a completed step with the same
     // output label.
+    //
+    // Some input labels may have no step that produces that label as an output.
+    // This are the original sources of data. To execute the protocol, the party
+    // will need to provide a configuration showing where these original data
+    // sources come from.
     map<string, string> input_labels = 3;
     map<string, string> output_labels = 4;
 
-    // TODO(world-federation-of-advertisers/experimental#81): add step details.
+    oneof step {
+      EncryptAndHashStep encrypt_and_hash = 7;
+      IntersectAndValidateStep intersect_and_validate = 8;
+      JoinAndDecryptStep join_and_decrypt = 9;
+      TableQueryStep table_query = 10;
```

Comment:
"Table Query" is kind of a weird name. Is "Query" enough or "LogQuery"?

Reply:
Discussed 1:1 -- since these are generic building blocks for protocols, the name kind of makes sense for something to look up entries in a list of key-value pairs based on key. We'll leave as is.